coalib/bearlib/aspects/__init__.py

```diff
@@ -0,0 +1,322 @@
+   True
+   >>> Test(a="", b="test").check_consistency()
+   False
+
+   :param members: The members to check for consistency.
+   """
+    def decorator(cls):
+        cls.check_consistency = (
+            lambda self: all(getattr(self, member) for member in members)
+        )
+
+        return cls
+
+    return decorator
+
+
+@generate_consistency_check('definition', 'cause', 'example',
+                            'example_language', 'importance_reason',
+                            'fix_suggestions')
+class AspectDocumentation:
+    """
+    This class contains documentation about an aspect described by the Aspect
+    class.
+    The documentation is consistent if all members are given:
+    >>> AspectDocumentation('defined', '', '', '').check_consistency()
+    False
+    >>> AspectDocumentation('definition', 'cause', 'example',
+    ...                     'example_language', 'importance',
+    ...                     'fix').check_consistency()
+    True
+    """
+
+    @enforce_signature
+    def __init__(self, definition: str='', cause: str='', example: str='',
+                 example_language: str='', importance_reason: str='',
+                 fix_suggestions: str=''):
+        """
+        Contains documentation for an aspect.
+        :param definition:        What is this about?
+        :param cause:             Information on how such a problem can happen.
+        :param example:           An example in a well known language.
+        :param example_language:  The language used for the example.
+        :param importance_reason: A reason why this aspect is important.
+        :param fix_suggestions:   Suggestions on how this can be fixed.
+        """
+        super().__init__()
+
+        self.definition = cleandoc(definition)
+        self.cause = cleandoc(cause)
+        self.example = cleandoc(example)
+        self.example_language = cleandoc(example_language)
+        self.importance_reason = cleandoc(importance_reason)
+        self.fix_suggestions = cleandoc(fix_suggestions)
+
+
+@generate_consistency_check('name', 'description', 'cast_type')
+class AspectSetting:
+    """
+    This class contains the specification about a setting that should be given
+    for an aspect.
+    """
+
+    class NoDefault:
+        pass
+
+    @enforce_signature
+    def __init__(self,
+                 name='',
+                 description='',
+                 cast_type=str,
+                 default=NoDefault):
+        """
+        Instantiates an ``AspectSetting`` object.
+
+        :param name:
+            Name of the setting.
+        :param description:
+            A description of what the setting is about and what it does.
+        :param cast_type:
+            The type to cast the user value into. For example, ``int`` will
+            convert and setting value as an integer.
+        :param default:
+            Default value for the setting.
+        """
+        self.name = name
+        self.description = description
```

Comment:
There's the example with `Root.Redundancy.Clone` - we create a setting called `max_clone_tokens`. Do you mean like a doctest for `AspectSetting`?